iam_role.go

```diff
@@ -220,6 +249,92 @@ func (c *Client) CreateIamTrustRole(roleName string, roleType string, trustArn s
 	return cr, nil
 }
 
+type UpdateRoleRequest struct {
+	RoleName *string `json:"roleName"`
+	Tags     *[]Tag  `json:"tags"`
+}
+
+type UpdateRoleResponse struct {
+	BaseResponse
+	RoleArn         *string `json:"roleArn"`
+	RoleName        *string `json:"roleName"`
+	BasicAuth       *bool   `json:"basicAuthUsed"`
+	Exists          *bool   `json:"roleExists"`
+	RoleIPArn       *string `json:"instanceProfileArn"`
+	MachineIdentity *bool   `json:"isMachineIdentity"`
+	Tags            *[]Tag  `json:"tags"`
+}
+
+type requestOp struct {
+	Operation string
+	ReqObj    []byte
+	Method    string
+	Endpoint  string
+	Response  interface{}
+}
+
+func (c *Client) makeIamRoleRequest(op *requestOp) error {
```

Comment:
A couple things here:

I don't think the options-object pattern is really necessary here since we aren't really having to worry about getting around method overloading not being supported in this case. This is just an internal function call so we don't need this function to be future-proofed for that type of extensibility. Probably a lot simpler to just do regular function syntax where it takes five parameters

That being said, I'm not sure this function actually adds any value since it looks like it just adds more complication without actually buying us anything. The logic for building and submitting an http request in go is pretty simple by itself. If we were going to wrap the logic for that in a reusable function though I would think we would need to make it more generic so that we can use it for all the other functions, but at the moment this PR is already quite big. Might be a better idea to match the style of the other functions for now and deferring this refactored approach to a later PR

Reply:
Agreed. This could be cleaner. we'll come back to it but for now I'll mirror the other APIs